4.0/PROTOCOL.md

```diff
@@ -484,4 +484,31 @@ When built-in load balancing mechanisms are disabled, the balancing is done by t
 
 ## Changes from version `3`
 
-!!TODO!!
+**INFO**
+- added `instanceID: string`
+- added `metadata: object` specific to each nodes
+
+**PING / PONG**
+- added messaged ID `id: string` on both message types
+
+**REQUEST**
+- `params` now is not required anymore 
+- removed `metrics` property and added `tracing: boolean` whether tracing events is needed
+- new `paramsType: enum` needed for schema-based serializers
+- new `caller: string` property for action name of the caller
+- new `seq: int32` to have better tracking of stream request
+
+**RESPONSE**
+- new `paramsType: enum` needed for schema-based serializers
```

Comment:
It's not `paramsType`, instead `dataType`

Reply:
oh yes